src/http/request_handler.rs

```diff
@@ -105,11 +105,31 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
     new_headers
 }
 
+pub async fn preflight(app_ctx: Arc<AppContext>, req: Request<Body>) -> Result<Response<Body>> {
+    let _whole_body = hyper::body::aggregate(req).await?;
+    let mut response = Response::builder().status(StatusCode::OK);
+
+    for (key, val) in &app_ctx.blueprint.server.response_headers {
```

Comment:
this is static info and list of CORS headers should be calculated upfront and just passed here instead of checking it on every request